loky/backend/popen_loky.py

```diff
@@ -195,13 +196,17 @@ def thread_is_spawning():
         semaphore_tracker._semaphore_tracker._fd = args.semaphore
 
     exitcode = 1
+
     try:
-        from_parent = os.fdopen(r, 'rb')
-        prep_data = pickle.load(from_parent)
-        spawn.prepare(prep_data)
-        process_obj = pickle.load(from_parent)
-        from_parent.close()
-        from_parent = None
+        with os.fdopen(r, 'rb') as from_parent:
+            process.current_process()._inheriting = True
+            try:
+                prep_data = pickle.load(from_parent)
+                spawn.prepare(prep_data)
+                process_obj = pickle.load(from_parent)
+            finally:
+                del process.current_process()._inheriting
```

Comment:
Previously, we did not set the `_inheriting` flag to `True`.
This flag is checked during the `_check_not_importing_main` and we raise an error when it is True.
This function is called when we are spawning a new process and we should not spawn a new process before we finished loading the current one.
This fixes the behavior by setting the flag while we load the process data.